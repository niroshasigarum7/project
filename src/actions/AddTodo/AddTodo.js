let nextTodoId = 0
export const todos = text => ({
    type: 'ADD_TODO',
    id: nextTodoId++,
    text
})