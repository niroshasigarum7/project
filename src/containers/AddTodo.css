.add-todo__container {
    margin: 20px 20px;
    padding-bottom: 20px;
    padding-top: 20px;
}

.button {
    padding: 15px 25px;
    font-size: 12px;
    text-align: center;
    cursor: pointer;
    outline: none;
    color: #fff;
    background-color: mediumvioletred;
    border: none;
    border-radius: 15px;
    box-shadow: 0 9px #999;
}

.input {
    height: 39px;
    border: none;
    border-radius: 15px;
    box-shadow: 0 9px mediumvioletred;
}