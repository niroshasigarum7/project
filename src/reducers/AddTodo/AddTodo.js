import { ADD_TODO, TOGGLE_TODO} from '../../actions/AddTodo/AddTodo';

const initialState = [
    {
        text: 'Use Redux',
        completed: false,
        id: 0
    }
]

const todos = (state = initialState, action) => {
    switch (action.type) {
        case ADD_TODO:
            return [
                {
                    id: state.reduce((maxId, todo) => Math.max(todo.id, maxId), -1) + 1,
                    completed: false,
                    text: action.text
                },
                ...state
            ]
        case TOGGLE_TODO:
            return state.map(todo =>
                todo.id === action.id
                ? { ...todo, completed: !todo.completed }
                : todo
            )
        default:
            return state
    }
}

export default todos