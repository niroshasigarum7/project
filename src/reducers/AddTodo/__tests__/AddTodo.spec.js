import reducer from '../AddTodo';
import * as types from '../../../actions/AddTodo/AddTodo';
import expect from 'expect';
/*
https://redux.js.org/recipes/writing-tests
https://medium.com/@netxm/testing-redux-reducers-with-jest-6653abbfe3e1
*/
const action = {
        type: types.ADD_TODO,
        text: 'Run the tests'
};
const initialState = [{
        text: 'Use Redux',
        completed: false,
        id: 0
}];
const expectedState = [
        {
                text: 'Run the tests',
                completed: false,
                id: 1
        },
        {
                text: 'Use Redux',
                completed: false,
                id: 0
        }
];
describe('AddTodo', () => {
        it('should test the initial state', () => {
                expect(reducer(undefined, action)).toEqual(initialState);
                expect(reducer(undefined, action)).toMatchSnapshot();
        });
        it('returns the expected or correct state', () => {
                expect(reducer(initialState, action)).toEqual(expectedState);
                expect(reducer(initialState, action)).toMatchSnapshot();
        });
        // it('should handle ADD_TODO', () => {
        //         const actionType = 'ADD_TODO';
        //         expect(reducer(initialState, actionType)).toEqual(expectedState);
        // });
});
        