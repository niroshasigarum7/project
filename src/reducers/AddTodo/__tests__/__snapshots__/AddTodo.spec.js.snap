// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo returns the expected or correct state 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Use Redux",
  },
]
`;

exports[`AddTodo should test the initial state 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Use Redux",
  },
]
`;
